### ISSUE TITLE
in /v2/op/update if entities is empty CB crashes

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/fiware-orion#2720]
in /v2/op/update if entities is empty CB crashes

#### CB version
```
  "version" : "1.5.0-next"
  "git_hash" : "d943fb6827fdee0d7e89501c343dc252f6757f60"
```

#### Request
```
curl http://qa-orion-fe-02:1026/v2/op/update -H "Content-Type: application/json" -d '{"entities": [], "actionType": "APPEND"}' 
```
#### Response
```
curl: (52) Empty reply from server
```

#### log
````
time=2016-11-25T13:07:27.918Z | lvl=DEBUG | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=rest.cpp[1123]:connectionTreat | msg=
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=rest.cpp[1125]:connectionTreat | msg=--------------------- Serving request POST /v2/op/update -----------------
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=rest.cpp[500]:httpHeaderGet | msg=HTTP Header:   User-Agent: curl/7.19.7 (x86_64-redhat-linux-gnu) libcurl/7.19.7 NSS/3.19.1 Basic ECC zlib/1.2.3 libidn/1.18 libssh2/1.4.2
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=rest.cpp[500]:httpHeaderGet | msg=HTTP Header:   Host: qa-orion-fe-02:1026
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=rest.cpp[500]:httpHeaderGet | msg=HTTP Header:   Accept: */*
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=rest.cpp[500]:httpHeaderGet | msg=HTTP Header:   Content-Type: application/json
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=rest.cpp[500]:httpHeaderGet | msg=HTTP Header:   Content-Length: 40
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=sem.cpp[276]:transSemTake | msg=transactionIdSet taking the 'trans' semaphore for 'changing the transaction id'
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=sem.cpp[297]:transSemTake | msg=transactionIdSet has the 'trans' semaphore
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=sem.cpp[373]:transSemGive | msg=transactionIdSet gives the 'trans' semaphore for 'changing the transaction id'
time=2016-11-25T13:07:27.919Z | lvl=INFO | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=pending | srv=pending | subsrv=pending | comp=Orion | op=logMsg.h[1808]:lmTransactionStart | msg=Starting transaction from 127.0.1.1:53972/v2/op/update
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=pending | comp=Orion | op=rest.cpp[1231]:connectionTreat | msg=Got 40 of payload of 40 bytes
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=ContextAttribute.cpp[157]:ContextAttribute | msg=Creating a ContextAttribute 1
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=ContextAttribute.cpp[157]:ContextAttribute | msg=Creating a ContextAttribute 1
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=ContextAttribute.cpp[157]:ContextAttribute | msg=Creating a ContextAttribute 1
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[533]:restService | msg=Parsing payload for URL '/v2/op/update', method 'POST', service vector index: 33
time=2016-11-25T13:07:27.919Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[535]:restService | msg=Parsing payload '{"entities": [], "actionType": "APPEND"}'
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[108]:payloadParse | msg=parsing data for service 'BatchUpdateRequest'. Method: 'POST'
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[109]:payloadParse | msg=outMimeType: JSON
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[134]:payloadParse | msg=result:      'OK'
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[135]:payloadParse | msg=outMimeType: 'JSON'
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[537]:restService | msg=payloadParse returns 'OK'
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=pending | subsrv=<default> | comp=Orion | op=RestService.cpp[559]:restService | msg=Treating service POST /v2/op/update
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=<default> | subsrv=<default> | comp=Orion | op=RestService.cpp[600]:restService | msg=tenant: ''
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=<default> | subsrv=<default> | comp=Orion | op=ContextElementVector.cpp[88]:present | msg=0 ContextElements
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=<default> | subsrv=<default> | comp=Orion | op=UpdateActionType.cpp[105]:present | msg=UpdateActionType: APPEND
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=<default> | subsrv=<default> | comp=Orion | op=sem.cpp[147]:reqSemTake | msg=mongoUpdateContext taking the 'req' semaphore for 'ngsi10 update request'
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=<default> | subsrv=<default> | comp=Orion | op=sem.cpp[168]:reqSemTake | msg=mongoUpdateContext has the 'req' semaphore
time=2016-11-25T13:07:27.920Z | lvl=DEBUG | corr=1d9f04d4-b310-11e6-86a3-005056a20feb | trans=1480079224-063-00000000002 | from=127.0.1.1 | srv=<default> | subsrv=<default> | comp=Orion | op=sem.cpp[353]:reqSemGive | msg=mongoUpdateContext gives the 'req' semaphore for 'ngsi10 update request'
```

**Comment by fgalan** [comment]
It also affedcts 1.4.5:

```
  "version" : "1.4.5_20161125104030",
  "git_hash" : "f9d071d919c2c081b8dd2f9a6eddf4e8cdd6689a",
```

**Comment by fgalan** [comment]
Fixed by PR #2731. Ready for re-test.

**Comment by iariasleon** [comment]
te-tested in 
```
  "version" : "1.5.0-next",
  "git_hash" : "efac9dc557735e3ef986aa0659c255efbdd0e341",
```

An error is returned but the http code is `200` instead of  `400`.
```
http code: 200 - OK
headers:
   Connection: Keep-Alive
   Content-Length: 61
   Content-Type: application/json
   Fiware-Correlator: 5ad2ca56-b57d-11e6-ad3b-005056a20feb
   Date: Mon, 28 Nov 2016 15:14:28 GMT
payload: {"error":"Bad Request","description":"empty entities vector"}

```

**Comment by fgalan** [comment]
Last @iariasleon comment fixed (supossely) by PR https://github.com/telefonicaid/fiware-orion/pull/2733

**Comment by iariasleon** [comment]
Re-tested in 
```
   "version" : "1.5.0-next",
  "git_hash" : "d988b35e836b13ef58b4ccf2543b35236c30a0da",
```

The error field is `Bad Request` instead of `BadRequest`
```
http code: 400 - Bad Request
headers:
   Connection: Keep-Alive
   Content-Length: 61
   Content-Type: application/json
   Fiware-Correlator: c389fffe-b589-11e6-90ec-005056a20feb
   Date: Mon, 28 Nov 2016 16:43:17 GMT
payload: {"error":"Bad Request","description":"empty entities vector"}
```

**Comment by fgalan** [comment]
Last try to fix this in PR https://github.com/telefonicaid/fiware-orion/pull/2736. I hope the last one ;) A la tercera va la vencida

**Comment by iariasleon** [comment]
LGTM
```
http code: 400 - Bad Request
headers:
   Connection: Keep-Alive
   Content-Length: 60
   Content-Type: application/json
   Fiware-Correlator: 346bfcf2-b60e-11e6-af27-005056a20feb
   Date: Tue, 29 Nov 2016 08:31:20 GMT
payload: {"error":"BadRequest","description":"empty entities vector"}
```

**Comment by fgalan** [comment]
Same fix for release/1.4.0 in PR https://github.com/telefonicaid/fiware-orion/pull/2746

**Comment by iariasleon** [comment]
Tested in CB version
```
  "version" : "1.4.5",
  "git_hash" : "11a097192823d7cf64575d161025abff5eaa3320",
```
LGTM
```
http code: 400 - Bad Request
headers:
   Connection: Keep-Alive
   Content-Length: 60
   Content-Type: application/json
   Fiware-Correlator: c0a6c678-b886-11e6-b42a-005056a20feb
   Date: Fri, 02 Dec 2016 11:59:17 GMT
payload: {"error":"BadRequest","description":"empty entities vector"}

```
